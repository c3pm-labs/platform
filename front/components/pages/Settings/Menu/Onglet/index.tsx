import { useState, useEffect } from 'react';
import {
  makeStyles, Typography,
} from '@material-ui/core';
import clsx from 'clsx';

import Options from '../utils';

const useStyles = makeStyles((theme) => ({
  container: {
    transition: '0.5s',
    height: '2.5em',
    minHeight: '2.5em',
    maxHeight: '2.5em',
    flex: 1,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '70%',
    border: 'none',
    backgroundColor: 'rgba(0,0,0,0)',
    padding: 0,
    '&:hover': {
      opacity: 0.8,
      cursor: 'pointer',
    },
    '&:focus': {
      outline: 0,
    },
  },
  greyOptions: {
    color: `${theme.palette.text.primary}`,
  },
  containerVerticalLine: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    Width: '4px',
    minWidth: '4px',
    maxWidth: '4px',
  },
  verticalLine: {
    borderRadius: `${theme.shape.borderRadius}px`,
    display: 'flex',
    height: '2.5em',
    border: 'solid',
    width: '0px',
  },
  verticalLineFocused: {
    borderWidth: '2px',
    borderColor: `${theme.palette.primary.main}`,
  },
  verticalLineBlur: {
    borderWidth: '0.5px',
    borderColor: `${theme.palette.text.disabled}`,
  },
  containerTile: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    Width: '90px',
    minWidth: '90px',
    maxWidth: '90px',
  },
}));

interface OngletsProps {
  currentFocus: Options;
  title: Options;
  setIsMenu: (newValue: boolean) => void;
  onClick: (newValue: Options) => void;
}

function Onglet(props: OngletsProps): JSX.Element {
  const {
    currentFocus,
    title,
    onClick,
    setIsMenu,
  } = props;
  const classes = useStyles();
  const [isFocus, setIsFocus] = useState(false);

  useEffect(() => {
    if (currentFocus === title) setIsFocus(true);
    else setIsFocus(false);
  }, [currentFocus, title]);

  const handleClassNameVerticalLine = () => (
    clsx((isFocus ? classes.verticalLineFocused : classes.verticalLineBlur), classes.verticalLine)
  );

  return (
    <button type="button" className={classes.container} onClick={(): void => { onClick(title); setIsMenu(false); }}>
      <div className={classes.containerVerticalLine}>
        <div className={handleClassNameVerticalLine()} />
      </div>
      <div className={classes.containerTile}>
        <Typography className={classes.greyOptions}>
          {title}
        </Typography>
      </div>
    </button>
  );
}

export default Onglet;
