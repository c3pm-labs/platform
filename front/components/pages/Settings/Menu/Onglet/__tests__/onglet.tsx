import { render } from '@testing-library/react';

import Options from '../../utils';
import Onglet from '../index';

test('Onglet', () => {
  const { container } = render(
    <Onglet
      currentFocus={'Profile' as Options}
      title={'Profile' as Options}
      // eslint-disable-next-line no-console
      setIsMenu={(value) => console.log('set this menu', value)}
      // eslint-disable-next-line no-console
      onClick={(value) => console.log('click', value)}
    />,
  );

  expect(container).toMatchSnapshot();
});
