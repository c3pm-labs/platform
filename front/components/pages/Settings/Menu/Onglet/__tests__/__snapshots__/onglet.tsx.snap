// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onglet 1`] = `
<div>
  <button
    class="makeStyles-container-1"
    type="button"
  >
    <div
      class="makeStyles-containerVerticalLine-3"
    >
      <div
        class="makeStyles-verticalLineFocused-5 makeStyles-verticalLine-4"
      />
    </div>
    <div
      class="makeStyles-containerTile-7"
    >
      <p
        class="MuiTypography-root makeStyles-greyOptions-2 MuiTypography-body1"
      >
        Profile
      </p>
    </div>
  </button>
</div>
`;
